### PR TITLE
Convince the theme mixin to ignore null values

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?
This PR helps us avoid writing code like this: 

image

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
more than happy to review...I need to learn what a mixin is first :)

File: packages/apollos-ui-kit/src/theme/mixins.tests.js [hunk 1]
```diff
@@ -55,4 +63,14 @@ describe('withThemeMixin', () => {
     );
     expect(tree).toMatchSnapshot();
   });
+  it('prunes null inputs', () => {
+    const tree = renderer.create(
+      <ThemeProvider>
+        <FlexedView>
+          <TypeExample />
```
Comment (redreceipt) [inline packages/apollos-ui-kit/src/theme/mixins.tests.js:70]: @vinnyjth did you mean to test `TypeExampleWithNullInputs` here?

Comment (vinnyjth) [reply packages/apollos-ui-kit/src/theme/mixins.tests.js:70]: 😅 Yes, thank you!

**Comment by vinnyjth** [comment]
@redreceipt Fixed.

File: packages/apollos-ui-kit/src/theme/mixins.tests.js (hunk 1 above)
Comment (vinnyjth) [reply packages/apollos-ui-kit/src/theme/mixins.tests.js:70]: Fixed.

**Comment by redreceipt** [comment]
@vinnyjth could you maybe drop a couple of comments in that code to explain your thoughts a bit? With two recursive functions and multiple one letter variables, Our future selves will thank you. Also, I don't follow what this is really doing at all but that's a personal problem 😂

**Comment by vinnyjth** [comment]
@redreceipt I added a story that demonstrates how Mixins should work. It seems straight forward, but prior to this PR the `secondary` color might have been `null`. (caviet - it might also have been merged in fine in this specific example
. The issues we saw implementing apps were often related to deeply nested color values, like `colors.text.secondary`, but i haven't taken the time to test) 

Simulator Screen Shot - iPhone 11 - 2020-08-18 at 09 33 38

This PR exists so we can get rid of blocks of code like this:

**Pull request merged**